cloud_experts_tutorials/rosa-mobb-dynamic-certificate-custom-domain-tutorial.adoc

```diff
@@ -0,0 +1,531 @@
++
+[source,yaml]
+----
+$  cat <<EOF | oc create -f -
+   apiVersion: operators.coreos.com/v1
+   kind: OperatorGroup
+   metadata:
+     generateName: cert-manager-
+     namespace: cert-manager
+   ---
+   apiVersion: operators.coreos.com/v1alpha1
+   kind: Subscription
+   metadata:
+     name: cert-manager
+     namespace: cert-manager
+   spec:
+     channel: stable
+     installPlanApproval: Automatic
+     name: cert-manager
+     source: community-operators
+     sourceNamespace: openshift-marketplace
+   EOF
+----
++
+[NOTE]
+====
+It will take a few minutes for this Operator to install and complete its set up.
+====
+. Wait for the `cert-manager` Operator to finish installing.
++
+The next steps depends on the successful installation of the operator. It is recommended that you login to your cluster console to confirm the success status of the `cert-manager` Operator, in the *Installed Operators* section.
++
+image::mobb-dynamic-cert-operator-install-status.png[]
+
+. Annotate the service account. This is to enable the AWS SDK client code running within the `cert-manager` pod to interact with the AWS STS service for temporary tokens, by assuming the IAM Role that was created in an earlier step. This is referred to as link:https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html[IRSA].
++
+[source,terminal]
+----
+$ oc annotate serviceaccount cert-manager -n cert-manager eks.amazonaws.com/role-arn=$ROLE
+----
++
+[NOTE]
+====
+Normally, after service account annotations, a restart of the pod is required. However, the next step automatically causes a restart of the pod.
+====
+. Update the CA truststore of the cert-manager pod. This step is usually not required. However, it was noticed that the cert-manager pod had difficulties in trusting the link:https://docs.aws.amazon.com/iam/index.html#sts[STS] and link:https://acme-v02.api.letsencrypt.org/[LetsEncrypt] endpoints.
++
+Using the following commands essentially downloads the CA chain of these endpoints, puts them into a ConfigMap, which then gets attached to the pod as a volume.
++
+Along with this step, you will also be link:https://cert-manager.io/docs/configuration/acme/dns01/#setting-nameservers-for-dns01-self-check[setting the NameServers that the cert-manager will use for DNS01 self-check].
++
+The volume attachment to the pod and the setting of `NameServers` will be done together by patching the cert-manager CSV resource to persist these changes to the cert-manager deployment. This will cause the rollout of a new deployment and restart of the `cert-manager` pod.
++
+[source,terminal]
+----
+$   openssl s_client -showcerts -verify 5 -connect sts.amazonaws.com:443 < /dev/null 2> /dev/null | awk '/BEGIN/,/END/{ if(/BEGIN/){a++}; print}'  > $SCRATCH_DIR/extra-ca.pem
+openssl s_client -showcerts -verify 5 -connect acme-v02.api.letsencrypt.org:443 < /dev/null 2> /dev/null | awk '/BEGIN/,/END/{ if(/BEGIN/){a++}; print}'  >> $SCRATCH_DIR/extra-ca.pem
```

Comment:
These are two separate openssl commands, so this line needs a `$` at the beginning